### DISCUSSION TITLE
Can't get extra pages to render

### DISCUSSION THREAD

**Discussion by chmcquee** [keycloakify/keycloakify#252] [Q&A]
I've followed the advanced starter Code to get extra pages to load.
Here's what my specific component looks like I'd like to render 
```
import { KcProps } from "keycloakify";
import Template from "keycloakify/lib/components/Template";
import { memo } from "react";
import type { I18n } from "../../i18n";
import type { KcContext } from "../../context/kcContext";
import Header from "../Header/Header";
type KcContext_ConfirmPasswordChange = Extract<
  KcContext,
  { pageId: "my-extra-page-1.ftl" }
>;
const ConfirmPasswordChange = memo(
  ({
    kcContext,
    i18n,
    ...props
  }: { kcContext: KcContext_ConfirmPasswordChange; i18n: I18n } & KcProps) => {
    const { url } = kcContext;

    const { msg } = i18n;

    return (
      <Template
        {...{ kcContext, i18n, ...props }}
        doFetchDefaultThemeResources={true}
        displayMessage={false}
        headerNode={<Header backLink={url.loginUrl} />}
        formNode={
          <>
            <h3 className="instructions">{msg("passwordConfirmed")}</h3>
            <p className="instructions">
              {msg("passwordConfirmedInstructions")}
            </p>
            <br />
            <a className="link" href={url.loginUrl}>
              {msg("loginRedirect")}
            </a>
          </>
        }
      />
    );
  }
);

export default ConfirmPasswordChange;
````

My kcContext is structured as follows: 
```
import { getKcContext } from "keycloakify/lib/getKcContext";

export const { kcContext } = getKcContext<
  // NOTE: register.ftl is deprecated in favor of register-user-profile.ftl
  // but let's say we use it anyway and have this plugin enabled: https://github.com/micedre/keycloak-mail-whitelisting
  // keycloak-mail-whitelisting define the non standard ftl global authorizedMailDomains, we declare it here.

  | { pageId: "my-extra-page-1.ftl"; }
  | { pageId: "login.ftl"; authorizedMailDomains: string[] }
>({
  // Determine what page to test (login or forgotten password page) for development.
  //mockPageId: "login.ftl",
  //mockPageId: "login-reset-password.ftl",
  //mockPageId:"login-verify-email.ftl",
 // mockPageId: "login-update-password.ftl",
  mockData: [
    {
      pageId: "login.ftl",
      locale: {
        //When we test the login page we do it in German
        currentLanguageTag: "de",
      },
    },
    {
      pageId: "login-reset-password.ftl",
      locale: {
        currentLanguageTag: "de",
      },
    },
    {
      pageId: "login-verify-email.ftl",
      locale: {
        currentLanguageTag: "de",
      },
    },
    {
      pageId: "login-update-password.ftl",
      locale: {
        currentLanguageTag: "de",
      },
    },
    {
      pageId: "my-extra-page-1.ftl",
      locale: {
        currentLanguageTag: "de",
      },
    },
  ],
});

export type KcContext = NonNullable<typeof kcContext>;
```

I also added  

```
"keycloakify": {
        "extraPages": [
            "my-extra-page-1.ftl"
        ]
    },
```

to the package JSON.

This is the error I'm getting
Type '{ doFetchDefaultThemeResources: true; displayMessage: false; headerNode: Element; formNode: Element; scripts: string | readonly string[] | undefined; stylesCommon: string | readonly string[] | undefined; ... 76 more ...; i18n: I18n<...>; }' is not assignable to type '{ kcContext: KcContextBase; i18n: I18n<MessageKeyBase>; }'.
  Types of property 'kcContext' are incompatible.
    Type '{ pageId: "my-extra-page-1.ftl"; } & Common' is not assignable to type 'KcContextBase'.
      Type '{ pageId: "my-extra-page-1.ftl"; } & Common' is not assignable to type 'IdpReviewUserProfile'.
        Property 'profile' is missing in type '{ pageId: "my-extra-page-1.ftl"; } & Common' but required in type '{ pageId: "idp-review-user-profile.ftl"; profile: { context: "IDP_REVIEW"; attributes: Attribute[]; attributesByName: Record<string, Attribute>; }; }'.ts(2322)
KcContextBase.d.ts(322, 9): 'profile' is declared here.

Thanks for your help  👍

**Comment by chmcquee** [comment]
@garronej am I missing something in the documentation?

**Comment by garronej** [comment]
Hi @chmcquee,  

1) Please take the time to format properly your questions. Use \`\`\`tsx or whatever language.  
2) Extra pages are only for pages that are known to Keycloak and exists but are not currently supported see: https://docs.keycloakify.dev/limitations#some-pages-still-have-the-default-theme.-why  
3) I've completely reworked the project starter and the documentation this week end, you might want to check it out:  
https://docs.keycloakify.dev/v6.x-greater-than-v6.12  
https://docs.keycloakify.dev/
